client/components/editor.vue

```diff
@@ -462,12 +440,37 @@
             sort: {
               version: 1,
             },
-          }).fetch();
+          }).fetch()
+            .map((x) => {
+              return Object.assign({}, x, {
+                step: Step.fromJSON(schema, x.step),
+              });
+            });
 
           if (newContents.length) {
-            this.$editorView.dispatch(collab.receiveTransaction(this.$editorView.state, _.pluck(newContents, 'step').map((step) => {
-              return Step.fromJSON(schema, step);
-            }), _.pluck(newContents, 'clientId')));
+            // Notify to other components that there is new content.
+            this.$emit('content-changed');
+            // Observe new confirmed steps.
+            if (collab.getVersion(this.$editorView.state) > 0) {
```

Comment:
I am a bit confused about the order of things here. Shouldn't we first call `collab.receiveTransaction`? Because that one might update and rebase current editor's content. And then we should inform that highlights have changed. And then we should emit `content-changed` so that comments get positioned properly based on latest state of the editor's content?

For example, currently you emit `highlight-added` and then you call `updateCursor`, but content for which this highlight has been added might not have been yet updated with other steps who got from the server, so cursor might not go where it should go? Or at least when `collab.receiveTransaction` is called it would move again?

Reply:
It is probably true that for highlights we do not need to know where exactly they are, but just that they were confirmed. And `collab.receiveTransaction` later on will update their position correctly. But it is still a bit strange order, don't you agree?

If this order is required, maybe add a comment why is this so.